Adapters/LdapAdapter.py

```diff
@@ -0,0 +1,347 @@
+        )
+        groups = []
+        unique_ids = []
+
+        for resource in resources:
+            if 'assignedGroupId' in resource:
+                for group_id in resource['assignedGroupId']:
+                    group = self._connector.search_for_entity(
+                        'perunGroupId=' + group_id + ',perunVoId=' +
+                        resource['perunVoId'] + ',' + self._ldap_base,
+                        '(objectClass=perunGroup)',
+                        ['perunGroupId', 'cn', 'perunUniqueGroupName',
+                         'perunVoId', 'uuid', 'description']
+                    )
+                    if group['perunGroupId'] not in unique_ids:
+                        groups.append(
+                            self._create_internal_representation_group(group)
+                        )
+
+                        unique_ids.append(group['perunGroupId'])
+
+        return groups
+
+    def get_user_attributes(self, user, attr_names):
+        user_id = user.id
+
+        return self._connector.search_for_entity(
+            'perunUserId=' + user_id + ',ou=People,' + self._ldap_base,
+            '(objectClass=perunUser)',
+            attr_names
+        )
+
+    def get_entityless_attribute(self, attr_name):
+        raise AdapterSkipException()
+
+    def get_vo_attributes(self, vo, attributes):
+        raise AdapterSkipException()
+
+    def get_facility_attribute(self, facility, attr_name):
+        raise AdapterSkipException()
+
+    def get_facility_by_rp_identifier(
+            self, rp_identifier,
+            rp_identifier_attr='perunFacilityAttr_entityID'):
+        attr_name = \
+            self._attribute_utils.get_ldap_attr_name(rp_identifier_attr) \
+            or "entityID"
+        ldap_result = self._connector.search_for_entity(
+            self._ldap_base,
+            '(&(objectClass=perunFacility)(' + attr_name + '=' +
+            rp_identifier + '))',
+            [self.PERUN_FACILITY_ID, self.CN, self.DESCRIPTION]
+        )
+        if not ldap_result:
+            self._logger.warning('perun:AdapterLdap: '
+                                 'No facility with entityID \'' +
+                                 rp_identifier + '\' found.')
+            return
+
+        return Facility(
+            ldap_result[self.PERUN_FACILITY_ID],
+            ldap_result[self.CN][0],
+            ldap_result[self.DESCRIPTION][0],
+            rp_identifier
+        )
+
+    def get_users_groups_on_facility(self, facility, user_id):
+
+        if not facility:
+            return []
+
+        id = facility.id
+
+        resources = self._connector.search_for_entities(
+            self._ldap_base,
+            '(&(objectClass=perunResource)(perunFacilityDn='
+            'perunFacilityId=' +
+            id + ',' + self._ldap_base + '))',
+            ['perunResourceId']
+        )
+
+        self._logger.debug('Resources - ' + json.dumps(str(resources)))

```

Comment:
I think, that str(obj) should be enough. (all occurences)
```suggestion
        self._logger.debug('Resources - ' +str(resources))

```